From dc2726e01ae7f10073937ed9f5ab0844e9becc75 Mon Sep 17 00:00:00 2001
From: Johan Gunnarsson <johan.gunnarsson@gmail.com>
Date: Tue, 8 Aug 2017 22:27:53 +0200
Subject: [PATCH] Fix typo in ufloat16
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

gcc 7 prints this warning without the patch:

In file included from /tmp/include/libtorrent/torrent_info.hpp:61:0,
                 from test.cc:38:
/tmp/include/libtorrent/policy.hpp: In member function ‘libtorrent::ufloat16& libtorrent::ufloat16::operator=(int)’:
/tmp/include/libtorrent/policy.hpp:87:16: warning: ‘<<’ in boolean context, did you mean ‘<’ ? [-Wint-in-bool-context]
     m_val = (v << 3) || (exp & 7);
             ~~~^~~~~
---
 include/libtorrent/policy.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/libtorrent/policy.hpp b/include/libtorrent/policy.hpp
index cebbc482d2..6f960c5422 100644
--- a/include/libtorrent/policy.hpp
+++ b/include/libtorrent/policy.hpp
@@ -84,7 +84,7 @@ namespace libtorrent
 					++exp;
 				}
 				TORRENT_ASSERT(exp <= 7);
-				m_val = (v << 3) || (exp & 7);
+				m_val = (v << 3) | (exp & 7);
 			}
 			return *this;
 		}
